internal/executor/cache/cache.go

```diff
@@ -9,6 +10,8 @@ import (
 	"path/filepath"
 )
 
+const bufSize = 10 * 1024 * 1024
```

Comment:
let's do 1Mb since agent streams in 1Mb chunks

```suggestion
const bufSize = 1024 * 1024 // 1Mb chunk size from the agent
```

Reply:
See 85c575b4d2975aff174a2e29b060424f34ced7cc.